minetest/minetest
# Add wear bar color API

Adds an api for modders to change the appearance of the wear bar on their tools.

Currently, the tool damage bar (called "wear bar" by dev wiki) color cannot be customized by modders, and is listed under hardcoded features. This PR fixes that by adding fields in item def as well as item metadata to change the appearance. (See lua_api.txt for api specifics)
Resolves #13325

## Summary
The `wear_bar` field in item definitions and item meta allows modders to customize the color of wear bars in a variety of ways.

## To do

This PR is Ready for Review.

## How to test

In the devtest game, grab a few of the pickaxes (5, 10, 50, and 100 use pickaxes are customized) and dig some stone, allowing their wear bars to appear. Note the progression through different colors (as described by the tool description) when the pickaxe takes more damage.
Additionally, there is a pickaxe called "Wear Bar Color Test," that when punched with will use a random color for its wear bar. You can use the  `/use_tool` command to damage the tool a bit in order to allow the wear bar to appear.

Screenshot demonstrating the different color possibilities:
![minetest_wear_bar_color_explanation](https://user-images.githubusercontent.com/77073745/225460982-7f44083d-843a-443d-8c3b-680cdf4990c6.png)
